docs/token-builder.md

```diff
@@ -0,0 +1,288 @@
+const tokenModel = new SecretToken(options);
+
+// tokenModel.toJSON() returns:
+// {
+//   name: 'myApiKey',
+//   type: 'secret',
+//   path: 'watson',
+//   skipCache: true,
+//   cacheOverride: 'xyz',
+// };
+
+```
+
+### HmacToken
+
+Replaces HMAC token with an HMAC signature generated from the token options
+
+O**ptions:**
+
+- **stringToSign**
+
+    String that represents the request & will be used when generating HMAC signature
+
+- **algorithm**
+
+    The hashing algorithm: `sha1` , `sha256`, `md5`
+
+    (for now we only support these 3, but can easily add more later if needed)
+
+- **secretName**
+
+    Name of the data source secret (ex: `watson`)
+
+- **encoding**
+
+    Once the signature is generated it needs to be encoded
+
+    The following encodings are supported:`hex`, `base64`,`base64url` ,`base64percent`
+
+    - `base64url` produces the same result as `base64` but in addition also replaces
+
+        `+` with `-` , `/` with `_` , and removes the trailing padding character `=`
+
+    - `base64percent` encodes the signature as `base64` and then also URI percent encodes it
+
+**Example:**
+
+```jsx
+const hmacOptions = {
+  name: 'hmac_sig',
+  cacheOverride: 'xyz',
+  skipCache: true,
+  options: {
+    stringToSign: 'some_message',
+    algorithm: 'sha1',
+    secretName: 'watson',
+    encoding: 'hex',
+  },
+};
+
+const tokenModel = new HmacToken(hmacOptions);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'hmac_sig',
+//   type: 'hmac',
+//   cacheOverride: 'xyz',
+//   skipCache: true,
+//   options: {
+//     stringToSign: 'some_message',
+//     algorithm: 'sha1',
+//     secretName: 'watson',
+//     encoding: 'hex',
+//   },
+// };
+```
+
+### Sha1Token
+
+Replaces token with a SHA-1 signature generated from the token options.
+
+**Options:**
+
+- text
+
+    The data that will be hashed to generate the signature
+
+- encoding
+
+    Used to encode the result of hash function.
+
+    Accepted values: `hex` or `base64`
+
+- tokens
+
+    An array of [Secret]() tokens that could be included in the `text`. When included, these tokens will be interpolated into the `text`
+
+**Example:**
+
+```jsx
+const tokens = [{ name: 'secureValue', type: 'secret', path: 'mySecretPath' }];
+const sha1Options = {
+  name: 'FavoriteBand',
+  cacheOverride: 'Movable Band',
+  options: {
+    text: 'my text',
+    encoding: 'hex',
+    tokens,
+  },
+  skipCache: true,
+};
+
+const tokenModel = new Sha1Token(sha1Options);
+
+tokenModel.toJSON() // returns:
+// {
+//   name: 'FavoriteBand',
+//   type: 'sha1',
+//   cacheOverride: 'Movable Band',
+//   skipCache: true,
+//   options: {
+//     text: 'my text',
+//     encoding: 'hex',
+//     tokens,
+//   },
+// };
+```
+
+## **Caching**
+
+Each token has a couple of properties for handling how the token is included when generating a cache key:
+
+- **skipCache** - if set to `true`, then the token will not be considered when caching requests
```

Comment:
suggestion: This is where I would show examples of how to provide these options when constructing a token.